readthedocs/projects/tasks.py

```diff
@@ -133,28 +133,27 @@ def sync_repo(self):
                 ),
             )
 
-        with self.project.repo_nonblockinglock(version=self.version):
```

Comment:
This lock was being shared by sync_repo and build_docs, I'm moving it to the top of the tasks instead.